Sources/MapboxDirections/RouteResponse.swift

```diff
@@ -57,4 +65,32 @@ extension RouteResponse: Codable {
             routes = nil
         }
     }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encodeIfPresent(code, forKey: .code)
+        try container.encodeIfPresent(message, forKey: .message)
+        try container.encodeIfPresent(error, forKey: .error)
+        try container.encodeIfPresent(uuid, forKey: .uuid)
+        try container.encodeIfPresent(routes, forKey: .routes)
+        try container.encodeIfPresent(waypoints, forKey: .waypoints)
+    }
+    
+    /**
+     Adds request- or response-specific information to each result in a response.
+     */
+    func postprocess(accessToken: String, apiEndpoint: URL, fetchStartDate: Date, responseEndDate: Date) {
+        guard let routes = self.routes else {
+            return
+        }
+        
+        for result in routes {
```

Comment:
I neglected to call it out specifically in #391, but the major reason for this refactoring is so that properties like `accessToken`, `routeIdentifier` (a misnomer), and `fetchStartDate` can live on the response object instead of each individual result object.

Moving these properties to the response object will likely have some effect on the navigation SDK, but I think it’s healthy that the SDK will need to keep track of a whole response and perhaps a route index within that response, not just an individual result.